kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/ExponentialJitterBackoffStrategy.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import java.time.Duration;
+import java.util.Random;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public class ExponentialJitterBackoffStrategy implements BackoffStrategy {
+
+    @NonNull
+    private final Duration initialDelay;
+    @NonNull
+    private final Duration maximumDelay;
+    private final double multiplier;
+    private final Random random;
+
+    public ExponentialJitterBackoffStrategy(@NonNull Duration initialDelay,
+                                            @NonNull Duration maximumDelay,
+                                            double multiplier,
+                                            Random random) {
+        this.initialDelay = initialDelay;
+        this.maximumDelay = maximumDelay;
+        this.multiplier = multiplier;
+        this.random = random;
+        if (multiplier < 1.0) {
+            throw new IllegalArgumentException("multiplier should not reduce the initial delay");
+        }
+
+    }
+
+    @Override
+    public Duration getDelay(int failures) {
+        if (failures == 0) {
+            return Duration.ZERO;
+        }
+        Duration backoff = getExponentialBackoff(failures);
+        backoff = backoff.plus(getRandomJitter(failures, backoff));
+        return backoff.compareTo(maximumDelay) < 0 ? backoff : maximumDelay;
+    }
+
+    private Duration getRandomJitter(int failures, Duration backoff) {
+        Duration prior = getExponentialBackoff(failures - 1);
+        long maxJitter = backoff.toMillis() - prior.toMillis();
+        return Duration.ofMillis(this.random.nextLong() % maxJitter);
```

Comment:
So, if this backoff is 1.5 seconds, the previous backoff was 1 second, then the jitter will be 1.5 seconds plus or minus 500ms.

So the jittering range gets exponentially bigger along with the backoff.

Another approach is to have a randomization factor, so that you say "jitter within +-10%" so a 1 second delay would be jittered to be randomized within 0.9s and 1.1s.